ql/src/java/org/apache/hadoop/hive/ql/hooks/HiveProtoLoggingHook.java

```diff
@@ -314,6 +314,7 @@ private void writeEvent(HiveHookEventProtoPartialBuilder builder) {
         try {
           if (eventPerFile) {
             if (!maybeRolloverWriterForDay()) {
+              IOUtils.closeQuietly(writer);
```

Comment:
Why not just call `maybeRolloverWriterForDay()`. That method makes sure that the writer is open and ready?
Same way as for the normal case.
Like:
```
maybeRolloverWriterForDay();
LOG.debug("Event per file enabled. New proto event file: {}", writer.getPath());
writer.writeProto(event);
IOUtils.closeQuietly(writer);
writer = null;
```